app/ts/TradingUI/Trading.tsx

```diff
@@ -311,18 +296,26 @@ export const Trading = ({ maybeReadClient, maybeWriteClient, universe, forkValue
 		}
 	}
 
+	useSignalEffectWithAbortOnChange([maybeReadClient, selectedMarket, isRouterDeployed] as const, (_abortSignal, ...params) => refreshMarketData(...params), console.error)
+
+	const checkIfRouterIsDeployed = async (maybeReadClient: ReadClient | undefined) => {
+		if (maybeReadClient === undefined) return
+		isRouterDeployed.deepValue = await isAugurConstantProductMarketRouterDeployed(maybeReadClient)
+	}
+
+	useSignalEffectWithAbortOnChange([maybeReadClient] as const, (_abortSignal, ...params) => checkIfRouterIsDeployed(...params), console.error)
+
 	const refreshData = async () => {
 		await refreshMarketData(maybeReadClient.deepValue, selectedMarket.deepValue, isRouterDeployed.deepValue)
 	}
 
-	useSignalEffect(() => { updateAccountSpecificSignals(maybeWriteClient.deepValue).catch(console.error) })
-
 	const updateAccountSpecificSignals = async (maybeWriteClient: WriteClient | undefined) => {
 		if (maybeWriteClient === undefined) return
 		const router = getAugurConstantProductMarketRouterAddress()
 		daiApprovedForRouter.deepValue = await getAllowanceErc20Token(maybeWriteClient, DAI_TOKEN_ADDRESS, maybeWriteClient.account.address, router)
 		sharesApprovedToRouter.deepValue = await isErc1155ApprovedForAll(maybeWriteClient, AUGUR_SHARE_TOKEN, maybeWriteClient.account.address, router)
 	}
+	useSignalEffectWithAbortOnChange([maybeWriteClient] as const, (_abortSignal, ...params) => updateAccountSpecificSignals(...params), console.error)
```

Comment:
You'll start using the abort later I assume?  Without using the abort, I'm not sure we gain much by switching.